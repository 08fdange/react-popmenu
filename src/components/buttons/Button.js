import React from 'react';
import PrimaryButton from './PrimaryButton';
import SecondaryButton from './SecondaryButton';
import TransparentButton from './TransparentButton';

const Button = (props) => {
  const { variant, ...rest } = props;

  switch(variant) {
    case 'transparent':
      return <TransparentButton {...rest} />;
    case 'secondary':
      return <SecondaryButton {...rest} />;
    case 'primary':
    default:
      return <PrimaryButton {...rest} />;
  };
};

export default Button;