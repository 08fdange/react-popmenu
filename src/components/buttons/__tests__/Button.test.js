import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from '../Button';
import Colors from '../../../theme/colors';

test('Button - renders correctly', () => {
  const tree = renderer.create(<Button>Test</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button - renders default props correctly', () => {
  const testRenderer = renderer.create(<Button>Test</Button>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.children).toBe('Test');
  expect(testInstance.props.variant).toBe(undefined);
});

test('PrimaryButton - renders properly', () => {
  const testRenderer = renderer.create(<Button variant='primary'>Primary</Button>);
  const tree = testRenderer.toJSON();
  const testInstance = testRenderer.root;
  expect(testInstance.props.variant).toBe('primary');
  expect(tree).toHaveStyleRule('background-color', Colors.Purple);
});

test('SecondaryButton - renders properly', () => {
  const testRenderer = renderer.create(<Button variant='secondary'>Secondary</Button>);
  const tree = testRenderer.toJSON();
  const testInstance = testRenderer.root;
  expect(testInstance.props.variant).toBe('secondary');
  expect(tree).toHaveStyleRule('background-color', Colors.White);
});

test('TransparentButton - renders properly', () => {
  const testRenderer = renderer.create(<Button variant='transparent'>Transparent</Button>);
  const tree = testRenderer.toJSON();
  const testInstance = testRenderer.root;
  expect(testInstance.props.variant).toBe('transparent');
  expect(tree).toHaveStyleRule('background-color', 'transparent')
});