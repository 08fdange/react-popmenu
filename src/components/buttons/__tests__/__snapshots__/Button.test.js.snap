// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - renders correctly 1`] = `
.c0 {
  color: #FFFFFF;
  background-color: #4532E6;
  display: inline-block;
  font-size: 1em;
  font-family: 'Raleway';
  padding: 8px 12px;
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  border-radius: 6px;
  cursor: pointer;
}

.c0:focus {
  outline: 0;
  box-shadow: inset 0 2px 25px rgba(0,0,0,0.5);
}

.c0:hover {
  background-color: #271b8b;
}

<button
  className="c0"
>
  Test
</button>
`;
